include/alpaka/rand/RandDefault.hpp

```diff
@@ -0,0 +1,198 @@
+        {
+        };
+
+        namespace distribution
+        {
+            namespace gpu
+            {
+                namespace detail
+                {
+                    template<typename TFloat>
+                    struct BitsType;
+
+                    template<>
+                    struct BitsType<float>
+                    {
+                        typedef std::uint32_t type;
+                    };
+                    template<>
+                    struct BitsType<double>
+                    {
+                        typedef std::uint64_t type;
+                    };
+                } // namespace detail
+
+                //! The GPU random number normal distribution.
+                template<typename T>
+                class UniformUint
+                {
+                public:
+                    UniformUint() = default;
+
+                    template<typename TEngine>
+                    ALPAKA_FN_HOST_ACC auto operator()(TEngine& engine) -> T
+                    {
+                        using BitsT = typename TEngine::result_type;
+                        T ret = 0;
+                        constexpr auto N = sizeof(T) / sizeof(BitsT);
+                        for(unsigned int a = 0; a < N; ++a)
+                        {
+                            ret ^= (static_cast<T>(engine()) << (sizeof(BitsT) * 8 * a));
+                        }
+                        return ret;
+                    }
+                };
+
+                //! The GPU random number uniform distribution.
+                template<typename T>
+                class UniformReal
+                {
+                    using BitsT = typename detail::BitsType<T>::type;
+
+                public:
+                    UniformReal() = default;
+
+                    template<typename TEngine>
+                    ALPAKA_FN_HOST_ACC auto operator()(TEngine& engine) -> T
+                    {
+                        constexpr BitsT limit = static_cast<BitsT>(1) << std::numeric_limits<T>::digits;
+                        const BitsT b = UniformUint<BitsT>()(engine);
+                        const auto ret = static_cast<T>(b & (limit - 1)) / limit;
+                        return ret;
+                    }
+                };
+
+                //! The GPU random number normal distribution.
+                template<typename Acc, typename T>
+                class NormalReal
+                {
+                    const Acc& m_acc;
+                    T m_buf = static_cast<T>(NAN);
+
+                public:
+                    NormalReal(const Acc& acc) : m_acc(acc)
+                    {
+                    }
+
+                    template<typename TEngine>
+                    ALPAKA_FN_HOST_ACC auto operator()(TEngine& engine) -> T
+                    {
+                        constexpr T sigma = 1., mu = 0.;
+                        if(std::isnan(m_buf))
```

Comment:
Can you call `std::isnan` in device code? I am pretty sure you must call an alpaka math routine here. And if it does not exist, create it.

Reply:
Yes. This works with NVHPC and also OpenMP traget compilers aim to allow math functions form the `std` namespace. It also work in CUDA.

But your are right, that we may not want to rely on that. Adding this to `alpaka::math` should also come with it friends like `isfinite`, `isinfinite`, ... (six in total). I should put this in front in a separate PR, then.

Reply:
Yes please move this into a separate PR :heart: